spec/index.bs

```diff
@@ -154,70 +168,31 @@ WebGPU's coordinate systems match DirectX and Metal's coordinate systems in grap
   - Window/present coordinate matches framebuffer coordinate.
   - UV of origin(0, 0) in texture coordinate represents the first texel (the lowest byte) in texture memory.
 
-Type Definitions {#type-definitions}
-============================
-
-<script type=idl>
-typedef unsigned long long GPUBufferSize;
-</script>
-
-## Colors and Vectors ## {#colors-and-vectors}
-
-<script type=idl>
-dictionary GPUColorDict {
-    required double r;
-    required double g;
-    required double b;
-    required double a;
-};
-typedef (sequence<double> or GPUColorDict) GPUColor;
-</script>
-
-Note: `double` is large enough to precisely hold 32-bit signed/unsigned
-integers and single-precision floats.
-
-<script type=idl>
-dictionary GPUOrigin2DDict {
-    unsigned long x = 0;
-    unsigned long y = 0;
-};
-typedef (sequence<unsigned long> or GPUOrigin2DDict) GPUOrigin2D;
-</script>
-
-<script type=idl>
-dictionary GPUOrigin3DDict {
-    unsigned long x = 0;
-    unsigned long y = 0;
-    unsigned long z = 0;
-};
-typedef (sequence<unsigned long> or GPUOrigin3DDict) GPUOrigin3D;
-</script>
-
-<script type=idl>
-dictionary GPUExtent3DDict {
-    required unsigned long width;
-    required unsigned long height;
-    required unsigned long depth;
-};
-typedef (sequence<unsigned long> or GPUExtent3DDict) GPUExtent3D;
-</script>
-
-<script type=idl>
-typedef sequence<any> GPUMappedBuffer;
-</script>
-
-{{GPUMappedBuffer}} is always a sequence of 2 elements, of types {{GPUBuffer}}
-and {{ArrayBuffer}}, respectively.
-
 ## Internal Objects ## {#webgpu-internal-objects}
 
-An <dfn dfn>internal object</dfn> is a non-exposed conceptual WebGPU object.
+An <dfn dfn>internal object</dfn> is a conceptual, non-exposed WebGPU object.
 [=Internal objects=] track the state of an API object and hold any underlying implementation.
 If the state of a particular [=internal object=] can change in parallel from multiple [=agents=],
 those changes are always atomic with respect to all [=agents=].
 
 Note: An "[=agent=]" refers to a JavaScript "thread" (i.e. main thread, or Web Worker).
 
+### Invalid Objects ### {#invalid-objects}
+
+An [=internal object=] object may be <dfn dfn>invalid</dfn>.
+It may become [=invalid=] during its lifetime, but it will never become valid again.
+
+<div class=note>
+    [=Invalid=] objects result from a number of situations, including:
+
+      - If there is an error in the creation of the object, it is immediately invalid.
```

Comment:
Does encoding a bad command on an encoder make it invalid?

Reply:
This would be better reserved for that section of the spec, but good question.

Reply:
Agree, not necessary to talk about it here since this is non-comprehensive and non-normative.

But probably yes? It doesn't make much difference between that and "internal field that says whether it's going to be invalid"